src/main/scala/uk/gov/nationalarchives/consignmentexport/Bagit.scala

```diff
@@ -1,35 +1,47 @@
 package uk.gov.nationalarchives.consignmentexport
 
+import java.nio.charset.Charset
 import java.nio.file.Path
 import java.util
 import java.util.UUID
 
 import cats.effect.IO
 import gov.loc.repository.bagit.creator.BagCreator
-import gov.loc.repository.bagit.domain.{Bag, Metadata}
+import gov.loc.repository.bagit.domain.{Bag, Metadata, Manifest}
 import gov.loc.repository.bagit.hash.{StandardSupportedAlgorithms, SupportedAlgorithm}
 import gov.loc.repository.bagit.verify.BagVerifier
+import gov.loc.repository.bagit.writer.ManifestWriter
 import io.chrisdavenport.log4cats.SelfAwareStructuredLogger
+import uk.gov.nationalarchives.consignmentexport.ChecksumCalculator.ChecksumFile
 import uk.gov.nationalarchives.consignmentexport.Utils._
 
 import scala.jdk.CollectionConverters._
 import scala.language.postfixOps
 
 class Bagit(bagInPlace: (Path, util.Collection[SupportedAlgorithm], Boolean, Metadata) => Bag,
-            validateBag: (Bag, Boolean) => Unit)(implicit val logger: SelfAwareStructuredLogger[IO]) {
+            validateBag: (Bag, Boolean) => Unit,
+            writeTagManifests: (util.Set[Manifest], Path, Path, Charset) => Unit
+           )(implicit val logger: SelfAwareStructuredLogger[IO]) {
 
-  def createBag(consignmentId: UUID, rootLocation: String, metadata: Metadata): IO[Unit] = for {
+    def createBag(consignmentId: UUID, rootLocation: String, metadata: Metadata): IO[Bag] = for {
     bag <- IO(bagInPlace(
       s"$rootLocation/$consignmentId".toPath,
       List(StandardSupportedAlgorithms.SHA256: SupportedAlgorithm).asJavaCollection,
       true,
       metadata))
     _ <- IO(validateBag(bag, true))
     _ <- logger.info(s"Bagit export complete for consignment $consignmentId")
-  } yield ()
+  } yield bag
+
+  def writeMetadataFilesToBag(bag: Bag, checksumFiles: List[ChecksumFile]): IO[Unit] = IO {
+    val fileToChecksumMap: util.Map[Path, String] = checksumFiles.map(f => f.file.toPath -> f.checksum).toMap.asJava
+    bag.getTagManifests.asScala.head.getFileToChecksumMap.putAll(fileToChecksumMap)
```

Comment:
Is this the standard way of updating the tag manifests? I'm just surprised they want the client to mutate the internal map. 🤔

Reply:
Java innit :smile: 
There's not a vast amount of documentation so there doesn't seem to be a standard way. I can change it to do something like this:

```
val tagManifests = bag.getTagManifests.asScala.head
    val tagManifestMap = tagManifests.getFileToChecksumMap
    tagManifestMap.putAll(fileToChecksumMap)
    val newSet = new util.HashSet[Manifest]()
    val newManifest = new Manifest(StandardSupportedAlgorithms.SHA256)
    newManifest.setFileToChecksumMap(tagManifestMap)
    newSet.add(newManifest)
    bag.setTagManifests(newSet)
```
I'm not sure if that's clearer but I don't mind. What do you think?

Reply:
Yeah, that's definitely not clearer. 😅 

It's just a shame they don't provide a method for adding new manifests, which would be possible - even in Java. 🙂